From b1def6bf76b355a5763388e1de410c699a6b1b47 Mon Sep 17 00:00:00 2001
From: Pavel Dubrova <pashadubrova@gmail.com>
Date: Sun, 6 Feb 2022 05:57:15 +0200
Subject: [PATCH 1/2] gralloc: Protect new buffer allocation support for legacy
 camera

The new buffer allocation mechanism introduced in commit 1ff363d
("gralloc: Add buffer allocation support for camera utils format.")
completely depends on newer CamX blobs.

Change-Id: I44e6cff7204e5302909156ba93ee688ef95de257
Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>
---
 gralloc/Android.bp   | 5 ++++-
 gralloc/gr_utils.cpp | 4 +++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gralloc/Android.bp b/gralloc/Android.bp
index 20043284..09726f27 100644
--- a/gralloc/Android.bp
+++ b/gralloc/Android.bp
@@ -25,7 +25,10 @@ cc_library_shared {
         "-Wno-sign-conversion",
         "-Wno-unused-parameter",
         "-Wno-unused-variable",
-    ] + select(soong_config_variable("qtidisplay", "target_no_raw10_custom_format"), {
+    ] + select(soong_config_variable("qtidisplay", "target_no_camera_custom_format"), {
+        "true": ["-DNO_CAMERA_CUSTOM_FORMAT"],
+        default: [],
+    }) + select(soong_config_variable("qtidisplay", "target_no_raw10_custom_format"), {
         "true": ["-DNO_RAW10_CUSTOM_FORMAT"],
         default: [],
     }) + select(soong_config_variable("qtidisplay", "target_uses_ycrcb_camera_preview"), {
diff --git a/gralloc/gr_utils.cpp b/gralloc/gr_utils.cpp
index 0047814c..5afe9811 100644
--- a/gralloc/gr_utils.cpp
+++ b/gralloc/gr_utils.cpp
@@ -187,7 +187,8 @@ bool IsCompressedRGBFormat(int format) {
   return false;
 }
 
-bool IsCameraCustomFormat(int format) {
+bool IsCameraCustomFormat([[maybe_unused]] int format) {
+#ifndef NO_CAMERA_CUSTOM_FORMAT
   switch (format) {
     case HAL_PIXEL_FORMAT_NV21_ZSL:
       if (CameraInfo::GetInstance() &&
@@ -211,6 +212,7 @@ bool IsCameraCustomFormat(int format) {
     default:
       break;
   }
+#endif
 
   return false;
 }
-- 
2.53.0

